ui/jetbrains/src/main/java/com/neueda/jetbrains/plugin/graphdb/jetbrains/ui/datasource/tree/TreeNodeModel.java

```diff
@@ -0,0 +1,95 @@
+package com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.datasource.tree;
+
+import com.neueda.jetbrains.plugin.graphdb.jetbrains.component.datasource.state.DataSourceApi;
+import com.neueda.jetbrains.plugin.graphdb.jetbrains.ui.datasource.metadata.dto.ContextMenu;
+
+import javax.swing.*;
+import java.util.Optional;
+
+public class TreeNodeModel implements TreeNodeModelApi {
```

Comment:
Let's move metadata specific logic to a "MetadataTreeNodeModel" (similar to how DataSourceTreeNodeModel works).

It's looks a bit odd when you extend from TreeNodeModel and already have some built-in logic that you override by doing method overrides.

I would even consider to create "RootTreeNodeModel" for tree root specifically.